packages/wallet/test/rollup-block-submitter.spec.ts

```diff
@@ -0,0 +1,415 @@
+  blockSubmitter.getLastQueued().should.equal(queued)
+  blockSubmitter.getLastConfirmed().should.equal(confirmed)
+
+  let expectedSubmitted: number
+  switch (submitted) {
+    case queued:
+      expectedSubmitted = queued
+      break
+    case confirmed:
+      expectedSubmitted = submitted + 1
+      break
+    default:
+      expectedSubmitted = submitted
+  }
+  blockSubmitter.getLastSubmitted().should.equal(expectedSubmitted)
+
+  if (expectedSubmitted !== submitted) {
+    // Check that block was submitted
+    dummyContract.blocksSubmitted.length.should.equal(1)
+    dummyContract.blocksSubmitted[0][0].should.equal(
+      blocks[0].transitions[0].stateRoot
+    )
+
+    // Check that last submitted was persisted
+    const lastSubmittedFromDB: number = await getLastSubmittedFromDB(db)
+    lastSubmittedFromDB.should.equal(expectedSubmitted)
+  }
+}
+
+describe('DefaultRollupBlockSubmitter', () => {
+  let blockSubmitter: RollupBlockSubmitter
+  let dummyContract: DummyContract
+  let db: DB
+  let rollupBlock: RollupBlock
+  let rollupBlock2: RollupBlock
+
+  beforeEach(async () => {
+    dummyContract = new DummyContract()
+    db = newInMemoryDB()
+    // @ts-ignore
+    blockSubmitter = new DefaultRollupBlockSubmitter(db, dummyContract)
+
+    rollupBlock = {
+      number: 1,
+      transitions: [
+        {
+          stateRoot: keccak256(
+            Buffer.from('some stuff to hash').toString('hex')
+          ),
+          senderSlotIndex: 0,
+          recipientSlotIndex: 1,
+          tokenType: 0,
+          amount: 10,
+          signature: await new DefaultSignatureProvider().sign('test'),
+        },
+      ],
+    }
+
+    rollupBlock2 = {
+      number: 2,
+      transitions: [
+        {
+          stateRoot: keccak256(
+            Buffer.from('different stuff to hash').toString('hex')
+          ),
+          senderSlotIndex: 1,
+          recipientSlotIndex: 0,
+          tokenType: 1,
+          amount: 100,
+          signature: await new DefaultSignatureProvider().sign('test'),
+        },
+      ],
+    }
+  })
+
+  describe('init()', () => {
+    it('should init without error when DB empty', async () => {
+      await initQueuedSubmittedConfirmed(
+        db,
+        blockSubmitter,
+        dummyContract,
+        0,
+        0,
+        0
+      )
+    })
+
+    it('should init without error when block one submitted but not confirmed', async () => {
+      await initQueuedSubmittedConfirmed(
+        db,
+        blockSubmitter,
+        dummyContract,
+        1,
+        1,
+        0,
+        [rollupBlock]
+      )
+    })
+
+    it('should init without error when block 2 submitted but not confirmed', async () => {
+      await initQueuedSubmittedConfirmed(
+        db,
+        blockSubmitter,
+        dummyContract,
+        2,
+        2,
+        0,
+        [rollupBlock, rollupBlock2]
+      )
+    })
+
+    it('should try to submit when one queued but not submitted', async () => {
```

Comment:
In theory, a more robust test would actually check that an attempted submission to L1 happens here, right?

Reply:
It does: https://github.com/plasma-group/pigi/pull/474/files/088cbabddece5a2cba2640574e890a47943ede68#diff-b05b44c2d5e0451d7661ca8d857b1a5fR87, although due to separation of concerns, that part is mocked, but we do assert that it calls the contract's `submitBlock(...)` function with the appropriate data.